api/v1alpha3/azurecluster_validation.go

```diff
@@ -53,6 +60,23 @@ func (c *AzureCluster) validateClusterSpec() field.ErrorList {
 		field.NewPath("spec").Child("networkSpec"))
 }
 
+// validateClusterName validates ClusterName
+func (c *AzureCluster) validateClusterName() field.ErrorList {
+	var allErrs field.ErrorList
+	if len(c.Name) > clusterNameMaxLenght {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("metadata").Child("Name"), c.Name,
+			fmt.Sprintf("Cluster Name longer than allowed length of %d characters", clusterNameMaxLenght)))
+	}
+	if success, _ := regexp.MatchString(clusterNameRegex, c.Name); !success {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("metadata").Child("Name"), c.Name,
+			fmt.Sprintf("Cluster Name doesn't match regex %s", clusterNameRegex)))
```

Comment:
nit: can we make this error message a little more user friendly and instead of printing the regex (or in addition to), list the constraints, ie. `Cluster name cannot contain '\/"'[]:|<>+=;,?*@&.', start with an underscore, or end with a hyphen.`

Reply:
changed the error message